makefiles/tools/esptool.inc.mk

```diff
@@ -28,22 +35,30 @@ else
   FFLAGS += --flash_mode $(FLASH_MODE) --flash_freq $(FLASH_FREQ)
   FFLAGS += $(BOOTLOADER_POS) $(RIOTCPU)/$(CPU)/bin/$(BOOTLOADER_BIN)
   FFLAGS += 0x8000 $(BINDIR)/partitions.bin
-  FFLAGS += 0x10000 $(FLASHFILE).bin
+  FFLAGS += 0x10000 $(FLASHFILE)
 endif
 
-.PHONY: esp-image-convert esp-qemu
-# prepare image to flash: convert .elf to .bin
-esp-image-convert:
+# This is the binary that ends up programmed in the flash.
+$(ELFFILE).bin: $(ELFFILE)
 	$(Q)$(ESPTOOL) --chip $(FLASH_CHIP) elf2image --flash_mode $(FLASH_MODE) \
 		--flash_size $(FLASH_SIZE)MB --flash_freq $(FLASH_FREQ) $(FLASH_OPTS) \
-		-o $(FLASHFILE).bin $(FLASHFILE)
+		-o $@ $<
+
+# Default partition table with no OTA. Can be replaced with a custom partition
+# table setting PARTITION_TABLE_CSV.
+PARTITION_TABLE_CSV ?= $(BINDIR)/partitions.csv
+
+$(BINDIR)/partitions.csv: $(FLASHFILE)
```

Comment:
```suggestion
$(BINDIR)/partitions.csv: $(ELFFILE).bin
```

Reply:
I believe this one is better to leave as FLASHFILE. If you are using the default partition table but want to use a different FLASHFILE you can do something like:

```make
FLASHFILE = $(BINDIR)/myflashfile.bin
$(BINDIR)/myflashfile.bin: $(ELFFILE).bin my-extra-static-data
  cat $^ > $@
```

The FFLAGS has `0x10000 $(FLASHFILE)` so the default partition table should match that. If we were to implement OTA at some point then we wouldn't be using this default partition table and this point doesn't matter; but I think this should match what FFLAGS has and using FLASHFILE gives a tiny bit more flexibility.

I changed the body of the rule to use `$<` to avoid writing this twice.

Reply:
Sure